receivers/factory.go

```diff
@@ -42,6 +44,10 @@ type FactoryConfig struct {
 	Logger   logging.Logger
 }
 
+func (fc *FactoryConfig) Decrypt(key, fallback string) string {
+	return fc.DecryptFunc(context.Background(), fc.Config.SecureSettings, key, fallback)
+}
```

Comment:
This is a wrapper to hide secure settings and context from notifier's config builder

Reply:
Why?

Reply:
Figured it out - this is a nice change - not sure if that `context.Background()` might cause us more hard than good in the end but let's roll with it for now.

Reply:
> Why?

To hide implementation details of secrets storage from config parser, and reduce the scope for functions NewConfig. The context will be enclosed upstream (in Grafana) where it makes sense because it provides a secret service that requires context.